syntax = "proto3";

service TransactionService {
    rpc TransactionAdd (TransactionAddRequest)
                        returns (TransactionAddReply) {}
    rpc TransactionGet (TransactionGetRequest)
                        returns (TransactionGetReply) {}
}

message Transaction {
    string owner_name = 1;
    string transaction_name = 2;
    double amount = 3;
    string date = 4;
}

message TransactionAddRequest {
    Transaction transaction = 1;
}

message TransactionAddReply {
    int32 status = 1;
}

message TransactionGetRequest {
    string owner_name = 1;
    string start_date = 2;
    string end_date = 3;
}


message TransactionGetReply {
    int32 status = 1;
    repeated Transaction transactions = 2;
}
